python/ray/tune/tune.py

```diff
@@ -685,18 +689,22 @@ class and registered trainables.
             f"Got '{type(config)}' instead."
         )
 
-    (
-        storage_path,
-        local_path,
-        remote_path,
-        sync_config,
-    ) = _resolve_and_validate_storage_path(
-        storage_path=storage_path, local_dir=local_dir, sync_config=sync_config
-    )
+    if _use_storage_context():
+        local_path, remote_path = None, None
```

Comment:
Note to self: I'm no longer modifying `storage_path` with the old resolution method.

Now, the exact `storage_path` specified by the user, which could be `None`, gets used to initialize `StorageContext`. See the constructor to see the new resolution logic.